.github/workflows/tag.yml

```diff
@@ -0,0 +1,43 @@
+name: Set git tag
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+    inputs:
+      pushDateTag:
+        description: "Push datestamp (production) tag"
+        type: boolean
+        required: false
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  packages: write
+
+jobs:
+  set_tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: "main"
+          fetch-depth: 0
+
+      - name: Set git tag
+        run: |
+          tag=$(date +'%Y%m%d')
+          git config --global user.name 'GitHub Actions'
+          git config --global user.email 'github-actions@users.noreply.github.com'
+
+          if git tag -l $tag; then
+            echo "Tag $tag already exists."
+          else
+            git tag $tag
+            git push origin $tag
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
We need to use the PAT (which I've configured in the secrets for this repository) for the push, so that github doesn't prevent the other workflow from triggering. See https://stackoverflow.com/questions/64076798/how-to-trigger-a-github-actions-workflow-on-push-to-another-branch/64078507#64078507